azkaban-common/src/main/java/azkaban/executor/ExecutableRamp.java

```diff
@@ -43,6 +50,7 @@
     private volatile int numOfIgnored = 0;
     private volatile boolean isPaused = false;
     private volatile int rampStage = 0;
+    private volatile long lastRampDownTime = 0;
```

Comment:
adding more comment. This variable is used to tracking the 'automatically ramp down' mechanism. Let's say, we change the Status to ramp up from Stage 1 to Stage 2. And, during the ramp up period, the system detected more workflow failure. Therefore, the system will be capable to automatically ramp down from Stage 2 to Stage 1. However, to avoid non-stop ramp down from Stage 1 to Stage 0, we need an extra condition to determine if we need to continue ramping down. lastRampDownTime is added on that purpose. When the ramp has been ramped down in 1 day (by comparing the saved timestamp in lastRampDownTime), the system will not be automatically ramped down again.

Reply:
I meant, please add comments in the code itself.